### DISCUSSION TITLE
Making screenshots in 'viewport' mode does not respect browser resolution

### DISCUSSION THREAD

**Discussion by wsbaser** [cypress-io/cypress#25246] [Questions and Help]
Hello,
I've created visual regression tests using cypress, but I'm having a problem with making screenshots for the areas of the page that have scrollbars. To solve the problem I've tried to increase browser resolution using the approach from this article https://www.cypress.io/blog/2021/03/01/generate-high-resolution-videos-and-screenshots/
It works. When I make screenshot in `'runner'` mode, then everything is fine. Window size is increased and I do not see the scrollbars.
![runner](https://user-images.githubusercontent.com/7660756/208895232-767e23fb-4570-4fa5-a405-c51e31d552ba.png)
But when I use
`cy.screenshot({capture:'viewport'})` 
then screenshot is taken using resolution 1920x885 and I still see the scrollbars.
![viewport (1)](https://user-images.githubusercontent.com/7660756/208895048-02fdf893-2475-478c-a1d4-e263ce9cecd1.png)

Is there a possible workaround for this problem? Basically I need to make a screenshot the same as in 'runner' mode, but without a runner.

Here is the repository to reproduce the issue
https://github.com/wsbaser/cypress-wikipedia

**Comment by portyu9** [comment]
Hi @wsbaser 

The browser window size and the Cypress application viewport are two different things.

In the reproduction, `before:browser:launch` changes the **outer browser window** to:

```js
3840 x 2160
```

while `cypress.json` sets the AUT viewport to:

```json
{
  "viewportWidth": 2000,
  "viewportHeight": 1320
}
```

`capture: 'runner'` captures the browser viewport, including the Cypress runner, so increasing the browser window affects that screenshot.

`capture: 'viewport'`, however, captures the **application under test in the Cypress viewport**. To make that larger, set the Cypress viewport itself:

```js
cy.viewport(3840, 2160)
cy.screenshot('page', {
  capture: 'viewport',
})
```

or configure it globally:

```js
export default defineConfig({
  viewportWidth: 3840,
  viewportHeight: 2160,
})
```

For headless runs it can still be useful to make the browser window large enough through `before:browser:launch`, but that does not replace `viewportWidth` / `viewportHeight`.

So for an app-only 4K screenshot, both dimensions need to be considered:

```text
browser window  -> large enough to host the runner/AUT
Cypress viewport -> 3840 x 2160 app viewport
capture          -> 'viewport'
```

This distinction was later explicitly confirmed by Cypress maintainers: resizing the browser does not resize the AUT; `cy.viewport()` or viewport configuration controls the application and its viewport screenshots.

Please mark this answer as accepted if it helped, thank you.
